### PR TITLE
PHOENIX-6767 Traversing through all the guideposts to prepare paralle…

### PR DESCRIPTION
**Pull request by chrajeshbabu**
…l scans is not required for salted tables when the query is point lookup

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java [hunk 1]
```diff
@@ -929,12 +929,19 @@ private static boolean clipKeyRangeBytes(RowKeySchema schema, int fieldIndex, in
      * @throws SQLException
      */
     private List<List<Scan>> getParallelScans(byte[] startKey, byte[] stopKey) throws SQLException {
-        List<HRegionLocation> regionLocations = getRegionBoundaries(scanGrouper);
-        List<byte[]> regionBoundaries = toBoundaries(regionLocations);
         ScanRanges scanRanges = context.getScanRanges();
         PTable table = getTable();
-        boolean isSalted = table.getBucketNum() != null;
         boolean isLocalIndex = table.getIndexType() == IndexType.LOCAL;
+        if(!isLocalIndex && scanRanges.isPointLookup()) {
```
Comment (stoty) [inline phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java:935]: nit:This is rightfully flagged by checkstyle

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/salted/SaltedTableIT.java [hunk 2]
```diff
@@ -62,6 +62,33 @@ public void testTableWithInvalidBucketNumber() throws Exception {
         }
     }
 
+    @Test
+    public void testPointLookupOnSaltedTable() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        try (Connection conn = DriverManager.getConnection(getUrl(), props);) {
+            String tableName = generateUniqueName();
+            String query = "create table " + tableName + " (a_integer integer not null "
+                    + "CONSTRAINT pk PRIMARY KEY (a_integer)) SALT_BUCKETS = 256";
```
Comment (stoty) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/salted/SaltedTableIT.java:71]: Doesn't creating this many regions slow down the test ?

**Review by stoty — APPROVED** [review]
+1 LGTM

**Review by gjacoby126 — APPROVED** [review]
+1

**Comment by gjacoby126** [comment]
@chrajeshbabu - looks like there are test failures in PhoenixTableLevelMetricsIT -- do the tests need to be updated with different metric values to assert on now that the query planning is more efficient in some cases?

File: phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java (hunk 1 above)
Comment (chrajeshbabu) [reply phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java]: Done.

File: phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixTableLevelMetricsIT.java [hunk 3]
```diff
@@ -549,6 +550,7 @@ private static void assertMetricValue(Metric m, MetricType checkType, long compa
         assertSelectQueryTableMetrics(tableName, false, 1, 0, 1, 0, 0, true, 0, 0, rs);
     }
 
+    @Ignore
```
[the remark this answers is not in the document]
Comment (chrajeshbabu) [reply phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixTableLevelMetricsIT.java:559]: done.

**Comment by chrajeshbabu** [comment]
> @chrajeshbabu - looks like there are test failures in PhoenixTableLevelMetricsIT -- do the tests need to be updated with different metric values to assert on now that the query planning is more efficient in some cases?

Since we are not calling get table regions API here custom connection queries don't have effect and ignored the tests.
@gjacoby126

**Comment by chrajeshbabu** [comment]
Thanks for reviews @stoty @gjacoby126 .

**Pull request merged**